zebra-state/src/service/finalized_state/disk_db.rs

```diff
@@ -264,8 +264,9 @@ impl ReadDisk for DiskDb {
         // Reading individual values from iterators does not seem to cause database hangs.
         self.db
             .iterator_cf(cf, rocksdb::IteratorMode::Start)
-            .next()
+            .next()?
             .map(|(key_bytes, value_bytes)| (K::from_bytes(key_bytes), V::from_bytes(value_bytes)))
+            .ok()
```

Comment:
Database iterator now returns `Result` so some changes are needed here and below.

Reply:
We can't replace database failures with `None`, because that returns failures as missing values.